### PR TITLE
sanity check fixes for chassis; and cache cleanup at the end of pytest session

### PR DESCRIPTION
**Pull request by sanmalho-git**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
- sanity check:
   - The BGP and interface sanity checks for multi-dut were changed to iterate through all the nodes in a multi-dut setup. However, for a T2 chassis, the multi-dut testbed contains a supervisor card as well, which doesn't have any BGP/PORT configuration. Thus, these checks fail for supervisor card.

- cache cleanup:
  - If we change the inventory file and re-run pytest, the old cached data for the inventory is used instead of picking up the changes. 
   
#### How did you do it?
- sanity_check:
    - Iterate through frontend_nodes of duthosts instead of all the nodes.
- cache cleanup:
    - Added fixture cleanup_cache_for_session that is called at the beginning of a session to remove the cached facts for all the DUTs in the testbed. This is not an automatic fixture, and is needed in the following scenarios:
       -  Running tests where some 'facts' about the DUT that get cached are changed.
       - Running tests/regression without running test_pretest which has a test to clean up cache (PR#2978)
       - Test case development phase to work out testbed information changes.

### CODE REVIEW

File: tests/conftest.py
Comment (yxieca) [inline tests/conftest.py:885]: Curious why cleanup happens after test instead of before test? If the cache is stale, won't it fail the test? If a test particularly changed something in the cache and don't want to keep the change for next test, then it should clean up after test, but in that case, the fixture shouldn't be autouse, each test should decide to use the fixture or not.

Comment (sanmalho-git) [reply tests/conftest.py]: Removed the cleanup_cache fixture as this cleanup is being added to test_pretest in PR #2978

**Comment by yxieca** [comment]
@sanmalho-git I think your cache removal code has legit use cases. I suggest: make it not auto use, so test case developer can temporarily reference it until the test case is stable. Change the clean up to before test so that the testcase don't have to fail once before cleanup.

File: tests/conftest.py
Comment (yxieca) [inline tests/conftest.py:884]: Sorry, should have asked this earlier. Can you add comments why this fixture is added. Since it is not autoused and it is not used in any current code, it might invite someone to 'clean it up'.

**Review by yxieca — APPROVED** [review]

File: tests/conftest.py
Comment (yxieca) [reply tests/conftest.py:887]: I added a comment for you. Please review/edit as you see fit. Otherwise, I'll merge when test is done again.

**Pull request merged**
